repos/system_upgrade/el7toel8/actors/checkmultiplepackageversions/actor.py

```diff
@@ -0,0 +1,49 @@
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.models import InstalledRPM
+from leapp.reporting import create_report, Report, Title, Summary, Severity, Flags, Remediation, RelatedResource
+from leapp.tags import IPUWorkflowTag, ChecksPhaseTag
+
+
+class CheckMultiplePackageVersions(Actor):
+    """
+    Check for problematic 32bit packages installed together with 64bit ones.
+
+    If a known problematic 32bit package is found, the upgrade will be inhibited with the detailed
+    report how to solve the problem if such a remedy exists.
+    """
+
+    name = 'multiple_package_versions'
+    consumes = (InstalledRPM,)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+    def process(self):
+        # package_name: remedy information
+        problem_package_map = {
+                'brlapi.i686': {'bugzilla': None},
+                'gnome-online-accounts-devel.i686': {
+                    'bugzilla': 'https://bugzilla.redhat.com/show_bug.cgi?id=1765627'},
+                'geocode-glib-devel.i686': {
+                    'bugzilla': 'https://bugzilla.redhat.com/show_bug.cgi?id=1765629'}}
+        actual_problems = []
+        for package in problem_package_map:
+            name, arch = package.split('.')
+            if has_package(InstalledRPM, name, arch) and has_package(InstalledRPM, name, 'x86_64'):
+                actual_problems.append(package)
+        if actual_problems:
+            remediation = ["yum", "remove", "-y"] + actual_problems
+            # generate RelatedResources for the report
+            related_resources = []
+            for package in actual_problems:
```

Comment:
this second loop can do all the work in the first loop where you create actual_problems, no need to loop through that again

Reply:
Done